apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -2148,23 +2149,48 @@ private function checkInheritedAttributes(IShare $share): void {
 				/** @var SharedStorage $storage */
 				$originalShare = $storage->getShare();
 				$inheritedAttributes = $originalShare->getAttributes();
-				// hide if hidden and also the current share enforces hide (can only be false if one share is false or user is owner)
-				$hideDownload = $hideDownload && $originalShare->getHideDownload();
-				// allow download if already allowed by previous share or when the current share allows downloading
-				$canDownload = $canDownload || $inheritedAttributes === null || $inheritedAttributes->getAttribute('permissions', 'download') !== false;
+				
+				// For federated shares: users can only be MORE restrictive, never LESS restrictive
+				// If parent has hideDownload=true, child MUST have hideDownload=true
+				$parentHidesDownload = $originalShare->getHideDownload();
+				
+				// Check if download permission is available from parent
+				$parentAllowsDownload = $inheritedAttributes === null || $inheritedAttributes->getAttribute('permissions', 'download') !== false;
+				
+				// Apply inheritance rules:
+				// 1. If parent hides download, child must hide download
+				// 2. If parent allows download, child can choose to hide or allow
+				// 3. If parent forbids download, child cannot allow download
+				if ($parentHidesDownload) {
+					$hideDownload = true; // Parent forces hide, child cannot override
+				} else {
+					$hideDownload = $userExplicitlySetHideDownload; // Respect user's choice when parent allows
+				}
```

Comment:
Seems more readable, and is consistent with the line below.

```suggestion
				$hideDownload = $parentHidesDownload ||  $userExplicitlySetHideDownload
```